Shared/Samples/Download preplanned map area/DownloadPreplannedMapAreaView.swift

```diff
@@ -16,53 +16,65 @@ import ArcGIS
 import SwiftUI
 
 struct DownloadPreplannedMapAreaView: View {
-    /// A Boolean value indicating whether to select a map.
-    @State private var isShowingSelectMapView = false
-    
     /// A Boolean value indicating whether to show delete alert.
     @State private var isShowingDeleteAlert = false
     
     /// The view model for this sample.
     @StateObject private var model = Model()
     
     var body: some View {
-        MapView(map: model.currentMap)
-            .overlay(alignment: .top) {
-                mapNameOverlay
-            }
-            .toolbar {
-                ToolbarItemGroup(placement: .bottomBar) {
-                    Spacer()
+        MapViewReader { mapViewProxy in
+            MapView(map: model.currentMap)
+                .onVisibleAreaChanged { visibleArea in
+                    guard model.offlineMapDidChange else { return }
+                    model.offlineMapDidChange = false
                     
-                    Button("Select Map") {
-                        isShowingSelectMapView.toggle()
-                    }
-                    .sheet(isPresented: $isShowingSelectMapView, detents: [.medium]) {
-                        MapPicker()
-                            .environmentObject(model)
-                    }
+                    let builder = EnvelopeBuilder(envelope: visibleArea.extent)
+                    builder.expand(by: 0.5)
```

Comment:
While this works, I think what Nick originally proposed is more appealing to me. We can set the initial viewpoint to a closer scale before the map is displayed. Maybe a private extension like Mark has done in the toolkit: https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/pull/286

Reply:
I applied this suggestion to set the initial viewpoint so that the map scale is zoomed in for each map, including the map initially presented rather than just the offline maps:

<img src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/117859673/35c1700f-415a-46d1-9720-ca3325210fb5" width="250">
<img src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/117859673/97f202f7-f62d-4c8b-8386-8b8b9a765e61" width="250">